src/provider/controller/IntentHandler.ts

```diff
@@ -47,16 +47,35 @@ export class IntentHandler {
     }
 
     private async raiseWithTarget(intent: IntentWithTarget): Promise<IntentResolution> {
-        const appInfo = await this._directory.getAppByName(intent.target);
+        let appInfo: Application|null;
 
-        if (!appInfo) {
-            throw new FDC3Error(ResolveError.TargetAppNotInDirectory, `No app in directory with name: ${intent.target}`);
-        }
+        const appWindow = this._model.findWindowByAppName(intent.target);
+
+        if (appWindow) {
+            // Target app is running -> ensure that intent is registered in model
+            if (!appWindow.hasIntentListener(intent.type)) {
+                throw new FDC3Error(
+                    ResolveError.TargetAppDoesNotHandleIntent,
+                    `App '${intent.target}' has not registered listeners for intent '${intent.type}'`
+                );
+            }
```

Comment:
Are we worried about the case where an App has multiple windows, and the listeners are on a different window to the one returned by findWindowByAppName?

Reply:
Yep @pjbroadbent and I have discussed about the apps-with-multiple-windows thing quite a bit, we decided to leave it for a future story, once we know a bit more about how the resolver deals with individual windows, and whether the concept of "target" may include window information or just app

Reply:
Cool cool. @pjbroadbent - just want to confirm this is as captured in Jira as it needs to be before I hit 'resolve conversation'

Reply:
I've added a story to cover the potential upcoming changes that we discussed (SERVICE-505), but wrote it in a "TBD if we do this or not" sense.

May be worth for the time being changing the above check to fetch all windows from the model and search for a matching window here(?).

Reply:
Scratch that. Was thinking the issue was specific to "intents with target", it's a bit more generic than that - and could do with a small story rather than building into this PR.

Have added SERVICE-506 to cover the change.